inference-engine/src/mkldnn_plugin/nodes/mkldnn_priorbox_node.cpp

```diff
@@ -0,0 +1,322 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_priorbox_node.h"
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <vector>
+
+#include <ie_parallel.hpp>
+#include <mkldnn_types.h>
+#include <ngraph/ngraph.hpp>
+#include <ngraph/opsets/opset1.hpp>
+
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+namespace {
+float clip_great(float x, float threshold) {
+    return x < threshold ? x : threshold;
+}
+
+float clip_less(float x, float threshold) {
+    return x > threshold ? x : threshold;
+}
+}
+
+bool MKLDNNPriorBoxNode::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto priorBox = std::dynamic_pointer_cast<const ngraph::opset1::PriorBox>(op);
+        if (!priorBox) {
+            errorMessage = "Only opset1 PriorBox operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNPriorBoxNode::MKLDNNPriorBoxNode(
+    const std::shared_ptr<ngraph::Node>& op,
+    const mkldnn::engine& eng,
+    MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    const auto priorBox = std::dynamic_pointer_cast<const ngraph::opset1::PriorBox>(op);
+    const ngraph::opset1::PriorBox::Attributes& attrs = priorBox->get_attrs();
+    offset = attrs.offset;
+    step = attrs.step;
+    min_size = attrs.min_size;
+    max_size = attrs.max_size;
+    flip = attrs.flip;
+    clip = attrs.clip;
+    scale_all_sizes = attrs.scale_all_sizes;
+    fixed_size = attrs.fixed_size;
+    fixed_ratio = attrs.fixed_ratio;
+    density = attrs.density;
+
+    bool exist;
+    aspect_ratio.push_back(1.0f);
+    for (float aspect_ratio_item : attrs.aspect_ratio) {
+        exist = false;
+
+        if (std::fabs(aspect_ratio_item) < std::numeric_limits<float>::epsilon()) {
+            THROW_IE_EXCEPTION << "aspect_ratio param can't be equal to zero";
```

Comment:
1. It is kind of deprecated macro. Please use IE_THROW instead. 
2. We usually try to make error message more meaningful for the user (at least add layer name and type). See https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/mkldnn_plugin/nodes/mkldnn_pad_node.cpp#L21 as an example.

Reply:
done, thanks